### DISCUSSION TITLE
Plugin: ForeignKey array?

### DISCUSSION THREAD

**Discussion by ZuperSero** [netbox-community/netbox#10046] [Q&A] [answered]
Hope someone can clarify this for me,
I want to make a form the can add multiple virtual machines as ForeignKeys. 
 
Model:
```python
class System(NetBoxModel):
    servers = models.ForeignKey(
        to='virtualization.VirtualMachine',
        on_delete=models.PROTECT,
        related_name='+',
        blank=True,
        null=True
    )
    #...
```
Form:
```python
class SystemForm(NetBoxModelForm):
    servers = DynamicModelMultipleChoiceField(
        queryset=VirtualMachine.objects.all()
    )
```
But with this approach its adding a list of virtualmachine elements. 
Is there a way to make the model key an array of ForeignKeys?

Thank you 😁

**Comment by jeremystretch** [comment, marked as answer]
A [foreign key](https://www.postgresql.org/docs/current/tutorial-fk.html) is a database field which references a value in some other table; it can only reference a single object. To represent relationships to multiple objects, Django provides [ManyToManyField](https://docs.djangoproject.com/en/4.1/ref/models/fields/#manytomanyfield), which works by constructing an intermediate table mapping objects on either side of the relationship.

**Reply by ZuperSero** [reply]
Thank you for the quick response 😀
So as far as I understand it takes the same parameters as foreign key? So i would be able to just switch it out? I tried but it proberbly needs a new database migration?
```python
    servers = models.ManyToManyField(
        to='virtualization.VirtualMachine',
        on_delete=models.PROTECT,
        related_name='+',
        blank=True,
        null=True
    )
```

**Comment by hagbarddenstore** [comment]
Use a ManyToMany field.
